src/Array/Extra.elm

```diff
@@ -708,13 +680,12 @@ insertAt index elementToInsert =
     --> True
 
 -}
-all : (element -> Bool) -> (Array element -> Bool)
-all isOkay =
-    \array ->
-        array
-            |> Array.foldl
-                (\element soFar -> soFar && isOkay element)
-                True
+all : (a -> Bool) -> Array a -> Bool
+all isOkay array =
+    array
+        |> Array.foldl
+            (\element soFar -> soFar && isOkay element)
+            True
```

Comment:
Is there a way to short-circuit these? Including `any` below?

Reply:
Not without access to the Array constructors (which we don't from here)

Reply:
(This is out of scope for this PR)

I think there may be some scope for benchmarking against `Array.toList >> List.any isOkay >> Array.fromList` for this reason.

Reply:
Prediction: toList/List.any/fromList is going to be more expensive almost always